### PR TITLE
Remove Vulkan SDK dependency with Vulkan headers and temporary fix for static assert in linux builds

### PR DESCRIPTION
**Pull request by ArtemisX64**

### CODE REVIEW

**Comment by Zopolis4** [comment]
vcpkg provides a vulkan-headers package, maybe use that instead of a submodule?

**Comment by Exzap** [comment]
> vcpkg provides a vulkan-headers package, maybe use that instead of a submodule?

Personally I'm fine with both here.

**Comment by ArtemisX64** [comment]
> vcpkg provides a vulkan-headers package, maybe use that instead of a submodule?

TBF, I am not a big fan of vcpkg. So, I never checked it. 
Also, I think the one provided by vcpkg is old. (1.2.203)

**Review by Exzap — COMMENTED** [review]
Looks good. Just unsure about the precompiled.h change, see comment

File: src/Common/precompiled.h [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#ifdef __clang__
+#define static_assert(...) static_assert(true, "")
```
[the remark this answers is not in the document]
Comment (ArtemisX64) [reply src/Common/precompiled.h:2]: > I assume this is a workaround for the standard library "U must be unsigned" assert? Was this committed by accident? If not, could you add a source comment to indicate what the workaround is for. I'm not super happy with flat out stubbing static_assert but it's better than telling people to edit system headers.

Yes. This is the workaround for now.

[the remark this answers is not in the document]
Comment (ArtemisX64) [reply src/Common/precompiled.h:4]: Ok added the comment

**Comment by ArtemisX64** [comment]
I think I found the reason for U must be unsigned" assert. @Exzap Please don't merge it now

**Comment by Exzap** [comment]
Thanks!

**Comment by Zopolis4** [comment]
> > vcpkg provides a vulkan-headers package, maybe use that instead of a submodule?
> 
> TBF, I am not a big fan of vcpkg. So, I never checked it. Also, I think the one provided by vcpkg is old. (1.2.203)

It's on 1.3.224 now, btw.

**Pull request merged**